src/lib/definitions/workflow.ts

```diff
@@ -56,1091 +99,64 @@ export interface Workflow {
-   */
-  id?: string;
-  /**
-   * State name
-   */
-  name?: string;
-  /**
-   * State type
-   */
-  type?: 'delay';
-  /**
-   * State end definition
-   */
-  end?: End;
-  /**
-   * State data filter
-   */
-  stateDataFilter?: Statedatafilter;
-  /**
-   * Amount of time (ISO 8601 format) to delay
-   */
-  timeDelay?: string;
-  /**
-   * States error handling and retries definitions
-   */
-  onErrors?: Error[];
-  /**
-   * Next transition of the workflow after the time delay
-   */
-  transition?: Transition;
-  /**
-   * Unique Name of a workflow state which is responsible for compensation of this state
-   */
-  compensatedBy?: string;
-  /**
-   * If true, this state is used to compensate another state. Default is false
-   */
-  usedForCompensation?: boolean;
-  metadata?: /* Metadata information */ Metadata;
-}
-export type End =
-  | boolean
-  | {
-      /**
-       * If true, completes all execution flows in the given workflow instance
-       */
-      terminate?: boolean;
-      /**
-       * Defines events that should be produced
-       */
-      produceEvents?: /* Produce an event and set its data */ Produceeventdef[];
-      /**
-       * If set to true, triggers workflow compensation. Default is false
-       */
-      compensate?: boolean;
-    };
-/**
- * Switch state data based condition
- */
-export interface Enddatacondition {
-  /**
-   * Data condition name
-   */
-  name?: string;
-  /**
-   * Workflow expression evaluated against state data. Must evaluate to true or false
-   */
-  condition: string;
-  /**
-   * Workflow end definition
-   */
-  end: End;
-  metadata?: /* Metadata information */ Metadata;
-}
-/**
- * Switch state data event condition
- */
-export interface Enddeventcondition {
-  /**
-   * Event condition name
-   */
-  name?: string;
-  /**
-   * References an unique event name in the defined workflow events
-   */
-  eventRef: string;
-  /**
-   * Explicit transition to end
-   */
-  end: End;
-  /**
-   * Event data filter definition
-   */
-  eventDataFilter?: Eventdatafilter;
-  metadata?: /* Metadata information */ Metadata;
-}
-export type Error =
-  | {
-      /**
-       * Domain-specific error name, or '*' to indicate all possible errors
-       */
-      error: string;
-      /**
-       * Error code. Can be used in addition to the name to help runtimes resolve to technical errors/exceptions. Should not be defined if error is set to '*'
-       */
-      code?: string;
-      /**
-       * References a unique name of a retry definition.
-       */
-      retryRef?: string;
-      transition: Transition;
-      end?: End;
+  states: States;
+
+  constructor(model: any) {
+    const result = { expressionLang: 'jq' } as Specification.Workflow;
+    Object.assign(this, result, model);
+
+    const functions = this.functions;
+    if (typeof functions === typeof []) {
+      this.functions = (functions as Function[]).map((f) => new Function(JSON.stringify(f))) as Functions;
     }
```

Comment:
I don't understand the copy of the reference of function neither the stringify in the `Function` constructor.

I thing I would go for:
```typescript
if (typeof model.functions !== typeof '') {
  this.functions = (model.functions as Function[]).map((f) => new Function(f)) as Functions;
}
```
but maybe I'm missing something?

Furthermore, I think this should ideally be done for all properties that are not primitive types, in all the definitions... That's the only way to properly hydrate the models I think.